packages/permissionless/accounts/etherspot/signerToEtherspotSmartAccount.ts

```diff
@@ -0,0 +1,479 @@
+ */
+const getAccountInitCode = async <entryPoint extends EntryPoint>({
+    entryPoint: entryPointAddress,
+    owner,
+    index,
+    ecdsaValidatorAddress,
+    bootstrapAddress
+}: {
+    entryPoint: entryPoint
+    owner: Address
+    index: bigint
+    ecdsaValidatorAddress: Address
+    bootstrapAddress: Address
+}): Promise<Hex> => {
+    if (!owner) throw new Error("Owner account not found")
+
+    // Build the account initialization data
+    const initialisationData = getInitialisationData({
+        entryPoint: entryPointAddress,
+        ecdsaValidatorAddress,
+        owner,
+        bootstrapAddress
+    })
+
+    return encodeFunctionData({
+        abi: createAccountAbi,
+        functionName: "createAccount",
+        args: [toHex(index, { size: 32 }), initialisationData]
+    })
+}
+
+/**
+ * Check the validity of an existing account address, or fetch the pre-deterministic account address for a etherspot smart wallet
+ * @param client
+ * @param owner
+ * @param entryPoint
+ * @param ecdsaValidatorAddress
+ * @param initCodeProvider
+ * @param deployedAccountAddress
+ * @param factoryAddress
+ */
+const getAccountAddress = async <
+    entryPoint extends EntryPoint,
+    TTransport extends Transport = Transport,
+    TChain extends Chain | undefined = Chain | undefined
+>({
+    client,
+    owner,
+    entryPoint: entryPointAddress,
+    ecdsaValidatorAddress,
+    bootstrapAddress,
+    factoryAddress,
+    index
+}: {
+    client: Client<TTransport, TChain>
+    owner: Address
+    factoryAddress: Address
+    entryPoint: entryPoint
+    ecdsaValidatorAddress: Address
+    bootstrapAddress: Address
+    index: bigint
+}): Promise<Address> => {
+    const factoryData = getInitialisationData({
+        entryPoint: entryPointAddress,
+        ecdsaValidatorAddress,
+        owner,
+        bootstrapAddress
+    })
+
+    return await readContract(client, {
+        address: factoryAddress,
+        abi: EtherspotWalletFactoryAbi,
+        functionName: "getAddress",
+        args: [toHex(index, { size: 32 }), factoryData]
+    })
+}
+
+export type SignerToEtherspotSmartAccountParameters<
+    entryPoint extends EntryPoint,
+    TSource extends string = string,
+    TAddress extends Address = Address
+> = Prettify<{
+    signer: SmartAccountSigner<TSource, TAddress>
+    entryPoint: entryPoint
+    address?: Address
+    index?: bigint
+    factoryAddress?: Address
+    metaFactoryAddress?: Address
+    accountLogicAddress?: Address
+    ecdsaValidatorAddress?: Address
+    deployedAccountAddress?: Address
+}>
+/**
+ * Build a etherspot smart account from a private key, that use the ECDSA signer behind the scene
+ * @param client
+ * @param privateKey
+ * @param entryPoint
+ * @param index
+ * @param factoryAddress
+ * @param accountLogicAddress
+ * @param ecdsaValidatorAddress
+ * @param deployedAccountAddress
+ */
+export async function signerToEtherspotSmartAccount<
+    entryPoint extends EntryPoint,
```

Comment:
Make this to `entryPoint extends ENTRYPOINT_ADDRESS_V07_TYPE` so that users are aware of the type of entryPoint they can pass.

Reply:
Changes done.